### PR TITLE
checking for card match function created in object

### PR DESCRIPTION
**Pull request by SarahMelki**
#25 created a class with methods to check for matching cards. There are a couple other methods that could be deleted based on who is tackling them like shuffleCards, canFlipCard and flipCard.

### CODE REVIEW

File: public/app.js [hunk 1]
```diff
@@ -1 +1,86 @@
...
+        this.active = true;
+        this.cardToCheck = null;
+        this.matchedCards = [];
+        this.toggleCards();
+        this.shuffleCards(this.cardsArray);
+    }
+
+    youWin() {
+        document.querySelector('add_a_class_to_display_win').classList.add('hidden');
+    }
```
[the remark this answers is not in the document]
Comment (JoeDravarol) [reply public/app.js:21]: Thanks for explaning, that make a lot of sense now!

**Review by r-Dev03 — CHANGES_REQUESTED** [review]
The shuffleCards function is backend functionality and has already been implemented, besides that LGTM!

**Comment by jericashall** [comment]
This looks good. Mire is working on the card flipping, so you can comment that section out. 

Can you explain this function
getCardType(card) {
        return card.getElementsByClassName('squirtle')[0].src;
    }
Should it always be checking 0 index instead of the cards index?

**Comment by SarahMelki** [comment]
> This looks good. Mire is working on the card flipping, so you can comment that section out.
> 
> Can you explain this function getCardType(card) { return card.getElementsByClassName('squirtle')[0].src; } Should it always be checking 0 index instead of the cards index?

Yup, so since it's only one card grabbed from the top of the array, (we need to get the value from the front face + using a class for this value, which I think after the changes should be 'dance' not 'squirtle') hence it's the card at index 0.

**Comment by SarahMelki** [comment]
I missed adding a function when I was saving my work from the draft file!! An important function for the mis-matched cards:

cardMismatch(card1, card2) {
        this.active = true;
        card1.classList.remove('visible');
        card2.classList.remove('visible');
        this.active = false;
    }

**Review by jericashall — CHANGES_REQUESTED** [review]
Remove the shuffle method since it's done on the backend and add the cardMismatch method that you comment. Otherwise looks good.

**Comment by SarahMelki** [comment]
I did the required changes but not sure where all the red 'checks have failed' are coming from. I followed all the steps though 😨

**Comment by jericashall** [comment]
Sarah I think you should close this pull request, make sure you've pulled the lasted committed changes into your repository and then open another pull request.

**Comment by MichaelPachec0** [comment]
> I did the required changes but not sure where all the red 'checks have failed' are coming from. I followed all the steps though 😨

Sorry for not being too active, I'll try and see what I can do on getting this PR ready to merge.

**Comment by MichaelPachec0** [comment]
@SarahMelki i have kept both your pr's code and current. this should be ready for review now.

**Review by jericashall — APPROVED** [review]

**Review by r-Dev03 — APPROVED** [review]
LGTM!

**Review by MichaelPachec0 — APPROVED** [review]
Alright, more than happy to commit  seeing the rest of the group approves.

**Pull request merged**